### ISSUE TITLE
Driver(PacketTooLarge) with latest alpha

### ISSUE THREAD

**Issue by jonhoo** [blackbeam/mysql_async#85]
With the latest alpha, after a `cargo update`, I now get `Driver(PacketTooLarge)` when trying to execute a "long" SQL query through `drop_query`. Specifically, this query:
```sql
CREATE TABLE `stories` (`id` int unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY, `always_null` int, `created_at` datetime, `user_id` int unsigned, `url` varchar(250) DEFAULT '', `title` varchar(150) DEFAULT '' NOT NULL, `description` mediumtext, `short_id` varchar(6) DEFAULT '' NOT NULL, `is_expired` tinyint(1) DEFAULT 0 NOT NULL, `is_moderated` tinyint(1) DEFAULT 0 NOT NULL, `markeddown_description` mediumtext, `story_cache` mediumtext, `merged_story_id` int, `unavailable_at` datetime, `twitter_id` varchar(20), `user_is_author` tinyint(1) DEFAULT 0,  INDEX `index_stories_on_created_at`  (`created_at`), fulltext INDEX `index_stories_on_description`  (`description`),   INDEX `is_idxes`  (`is_expired`, `is_moderated`),  INDEX `index_stories_on_is_expired`  (`is_expired`),  INDEX `index_stories_on_is_moderated`  (`is_moderated`),  INDEX `index_stories_on_merged_story_id`  (`merged_story_id`), UNIQUE INDEX `unique_short_id`  (`short_id`), fulltext INDEX `index_stories_on_story_cache`  (`story_cache`), fulltext INDEX `index_stories_on_title`  (`title`),  INDEX `index_stories_on_twitter_id`  (`twitter_id`),  INDEX `url`  (`url`(191)),  INDEX `index_stories_on_user_id`  (`user_id`)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
```
This worked fine in the past.

**Comment by jonhoo** [comment]
Err, looks like I was using an old alpha. Let me try again with the new alpha and re-open if it's still an issue.

**Comment by jonhoo** [comment]
Yup, seems like this is happening even with `alpha.4`!

**Comment by blackbeam** [comment]
Hi.
I can't reproduce this in my test environment.
Which mysql version do you use? What `SELECT @@max_allowed_packet` returns?

**Comment by jonhoo** [comment]
Ah, I see what happened here. I'm using [`msql-srv`](https://docs.rs/msql-srv/) as the server, which sends 1024 as the `max_allowed_packet`. In the past, a bug in `mysql_common` meant that that value was ignored (I think), but now that that has been fixed, the value it returns is suddenly respected, which causes `mysql_async` to (correctly) complain about the max packet size. Ignore more :sweat_smile:

**Comment by jonhoo** [comment]
Hmm, although it doesn't happen with `mysql_async 0.20.0` and `mysql_common 0.19.2`...

**Comment by jonhoo** [comment]
It does happen with `mysql 17.0.0` though, so looks like this is probably just `mysql_async 0.20.0` not correctly checking `max_allowed_packet`.
